flutter-studio/src/io/flutter/android/AndroidModuleLibraryManager.java

```diff
@@ -0,0 +1,198 @@
+    Library[] androidProjectLibraries = androidProjectLibraryTable.getLibraries();
+    if (androidProjectLibraries.length == 0) {
+      LOG.error("Gradle sync was incomplete -- no Android libraries found");
+      return null;
+    }
+    HashSet<String> urls = new HashSet<>();
+    for (Library refLibrary : androidProjectLibraries) {
+      urls.addAll(Arrays.asList(refLibrary.getRootProvider().getUrls(OrderRootType.CLASSES)));
+    }
+    updateLibraryContent(urls);
+    return null;
+  }
+
+  @NotNull
+  @Override
+  protected String getLibraryName() {
+    return LIBRARY_NAME + " for " + getProject().getName();
+  }
+
+  @NotNull
+  @Override
+  protected PersistentLibraryKind<AndroidModuleLibraryProperties> getLibraryKind() {
+    return LIBRARY_KIND;
+  }
+
+  private void scheduleUpdate() {
+    if (isUpdating.get()) {
+      return;
+    }
+
+    final Runnable runnable = this::updateAndroidLibraries;
+    DumbService.getInstance(getProject()).smartInvokeLater(runnable, ModalityState.NON_MODAL);
+  }
+
+  private void updateAndroidLibraries() {
+    if (!isUpdating.compareAndSet(false, true)) {
+      return;
+    }
+    update();
+  }
+
+  private void doGradleSync(Project flutterProject, Function<Project, Void> callback) {
+    // TODO(messick): Collect URLs for all Android modules, including those within plugins.
+    VirtualFile dir = flutterProject.getBaseDir().findChild("android");
+    assert (dir != null);
+    EmbeddedAndroidProject androidProject = new EmbeddedAndroidProject(FileUtilRt.toSystemIndependentName(dir.getPath()), null);
+    androidProject.init();
+
+    GradleSyncListener listener = new GradleSyncListener() {
+      @SuppressWarnings("override")
+      public void syncTaskCreated(@NotNull Project project, @NotNull GradleSyncInvoker.Request request) {}
+
+      @Override
+      public void syncStarted(@NotNull Project project, boolean skipped, boolean sourceGenerationRequested) {}
+
+      @Override
+      public void setupStarted(@NotNull Project project) {}
+
+      @Override
+      public void syncSucceeded(@NotNull Project project) {
+        callback.apply(androidProject);
+        isUpdating.set(false);
+      }
+
+      @Override
+      public void syncFailed(@NotNull Project project, @NotNull String errorMessage) {
+        isUpdating.set(false);
+      }
+
+      @Override
+      public void syncSkipped(@NotNull Project project) {
+        isUpdating.set(false);
+      }
+    };
+
+    GradleSyncInvoker.Request request = GradleSyncInvoker.Request.projectLoaded();
+    request.runInBackground = true;
+    GradleSyncInvoker gradleSyncInvoker = ServiceManager.getService(GradleSyncInvoker.class);
+    gradleSyncInvoker.requestProjectSync(androidProject, request, listener);
+  }
+
+  @NotNull
+  public static AndroidModuleLibraryManager getInstance(@NotNull final Project project) {
+    return ServiceManager.getService(project, AndroidModuleLibraryManager.class);
+  }
+
+  public static void startWatching() {
```

Comment:
It looks like this will only watch projects that are open at IntelliJ launch?

Reply:
This runs each time a project is opened.

Reply:
This is one of those "What was I thinking?" moments. I removed the loop from `startWatching()` and added the Project as a parameter. Thanks for catching that!